django/http/request.py

```diff
@@ -106,6 +106,11 @@ def _get_raw_host(self):
         if settings.USE_X_FORWARDED_HOST and (
                 'HTTP_X_FORWARDED_HOST' in self.META):
             host = self.META['HTTP_X_FORWARDED_HOST']
+            if settings.USE_X_FORWARDED_PORT and (
+                    'HTTP_X_FORWARDED_PORT' in self.META):
+                server_port = self.META['HTTP_X_FORWARDED_PORT']
+                if server_port != ('443' if self.is_secure() else '80'):
```

Comment:
not very DRY, but I'm trying not to be overly clever

Reply:
Couldn't you just call `self.get_port` here to get the `server_port`? The rest of the code looks solid

Reply:
Strike that, we cannot use `get_port` since it would also return the `SERVER_PORT` which would be wrong. Your code currently breaks when you set `HTTP_X_FORWARDED_HOST` to `host:port` and `HTTP_X_FORWARDED_PORT` to `port`. Maybe just add an `and ':' not in host` to your if to cover that?

Reply:
My interpretation of https://docs.djangoproject.com/en/3.0/ref/settings/#use-x-forwarded-host is that you should not use `USE_X_FORWARDED_PORT` if you are including the port number in the `X-Forwarded-Host` header. 

I guess I could raise a configuration error if that is the case...

Reply:
Would it be more appropriate to throw an error if the port is included in `HTTP_X_FORWARDED_HOST` or to use the `HTTP_X_FORWARDED_PORT` which would make an assumption about that value taking precedent if it exists?

Reply:
The way I read the docs, it means that if `X-Forwarded-Host` contains a port already we should not attach `X-Forwarded-Port`. So imo we are left with two options:

 * Throw an error if `USE_X_FORWARDED_PORT` is set to `True` and the forwarded host already contains one.
 * Simply ignore `HTTP_X_FORWARDED_PORT` if the forwarded host contains a port.

I am slightly leaning towards option 1) if it doesn't make the code to ugly (such things are always hard to debug if they don't raise errors)